Explorer/Assets/DCL/Passport/Modules/UserDetailedInfo_PassportModuleController.cs

```diff
@@ -0,0 +1,664 @@
+            view.CancelInfoButton.gameObject.SetActive(!isSaving);
+            view.SaveInfoButton.gameObject.SetActive(!isSaving);
+            view.DescriptionForEditMode.interactable = !isSaving;
+            view.SaveLinksButton.interactable = !isSaving;
+
+            foreach (var additionalInfoForEdition in instantiatedAdditionalFieldsForEdition)
+                additionalInfoForEdition.SetAsInteractable(!isSaving);
+        }
+
+        private void SetLinksSectionAsEditionMode(bool isEditMode)
+        {
+            SetLinksSectionAsSavingStatus(false);
+
+            foreach (var editionObj in view.LinksEditionObjects)
+                editionObj.SetActive(isEditMode);
+
+            foreach (var readOnlyObj in view.LinksReadOnlyObjects)
+                readOnlyObj.SetActive(!isEditMode);
+
+            if (isEditMode)
+            {
+                ClearLinksForEdition();
+                foreach (var link in instantiatedLinks)
+                    AddLink(link.Id, link.Title.text, link.Url, true);
+
+                SetNewLinkButtonActive(instantiatedLinksForEdition.Count < LINKS_MAX_AMOUNT);
+            }
+            else
+            {
+                view.LinksContainer.gameObject.SetActive(currentProfile.Links is { Count: > 0 });
+                saveLinksCts.SafeCancelAndDispose();
+            }
+
+            LayoutRebuilder.ForceRebuildLayoutImmediate(view.MainContainer);
+        }
+
+        private void CreateNewLink(string title, string url)
+        {
+            AddLink(Guid.NewGuid().ToString(), title, url, true);
+            LayoutRebuilder.ForceRebuildLayoutImmediate(view.MainContainer);
+        }
+
+        private void RemoveLink(Link_PassportFieldView linkToRemove)
+        {
+            var indexToRemove = 0;
+            foreach (var link in instantiatedLinksForEdition)
+            {
+                if (link.Id == linkToRemove.Id)
+                    break;
+
+                indexToRemove++;
+            }
+
+            if (indexToRemove >= instantiatedLinksForEdition.Count)
+                return;
+
+            linksPoolForEdition.Release(linkToRemove);
+            instantiatedLinksForEdition.RemoveAt(indexToRemove);
+            SetNewLinkButtonActive(true);
+        }
+
+        private void SetNewLinkButtonActive(bool isActive)
+        {
+            view.AddNewLinkButton.gameObject.SetActive(isActive);
+
+            if (isActive)
+                view.AddNewLinkButton.transform.SetAsLastSibling();
+        }
+
+        private void SaveLinksSection()
+        {
+            saveLinksCts = saveLinksCts.SafeRestart();
+            SaveLinksAsync(saveLinksCts.Token).Forget();
+            return;
+
+            async UniTaskVoid SaveLinksAsync(CancellationToken ct)
+            {
+                SetLinksSectionAsSavingStatus(true);
+
+                if (currentProfile.Links == null)
+                    currentProfile.Links = new List<LinkJsonDto>();
```

Comment:
I am not entirely sure that `Links` belong to the `Profile` as they are only used in the passport and can be reused for different profiles in the controller.

Additionally you violate encapsulation assigning them externally that can lead to a mess

Reply:
Done: https://github.com/decentraland/unity-explorer/pull/1277/commits/a7b5dbfc8f080b49a2e6367ab85a26e532dd0f90